src/book.py

```diff
@@ -0,0 +1,224 @@
+        except AttributeError:  # re.search returned None
+            start_pos = 0
+        try:
+            end_pos = re.search(rex_end, data).span()[0]
+        except AttributeError:  # re.search returned None
+            end_pos = None
+        meta_data = data[:start_pos]
+        text_of_book = data[start_pos:end_pos]
+
+        self._text = text_of_book
+        if clean:
+            self.clean(inplace=True)
+        self.start_pos = start_pos
+        self.end_pos = end_pos
+        self._meta = meta or meta_data
+        if author is None:
+            try:
+                self._author = re.search(
+                    r"(?<=Author: )[\w\s]+(?=\n)", meta_data).group().strip()
+            except AttributeError:
+                self._author = None
+        else:
+            self._author = author
+        if title is None:
+            try:
+                self._title = re.search(
+                    r"(?<=Title: )[\w\s]+(?=\n)", meta_data).group().strip()
+            except AttributeError:
+                self._title = None
+        else:
+            self._title = None
+
+    def clean(self, lemmatize=True, deromanize=False, lemma_pos='v', inplace=False):
+        """Cleans the full text
+
+        Returns a new BookText object with cleaned text derived 
+        from the full text
+
+        lemmatize (default True): cleaned data will be lemmatized
+        deromanize (default False): roman numerals will be removed from cleaned data
+        lemma_pos (default 'v'): position variable for lemmatizer
+            if lemma_pos='all', will run through all lemmatization types for each word
+        inplace (default False): replaces the text of the object with cleaned text
+        """
+        cleaned = self._text
+
+        garbage = '\ufeff|â€™|â€"|â€œ|â€˜|â€\x9d|â€œi|-|â€'
+        cleaned = re.sub(garbage, '', cleaned)
+
+        if lemmatize:
+            WNLemma = WordNetLemmatizer()
+            # The Lemmatizer only takes single words as inputs, so we need to
+            # break our text into individual words
+            tokens = word_tokenize(cleaned)
+            if lemma_pos != 'all':
+                lemmatized = [WNLemma.lemmatize(
+                    token, pos=lemma_pos) for token in tokens]
+            else:
+                lemmatized = [WNLemma.lemmatize(WNLemma.lemmatize(
+                    WNLemma.lemmatize(token, pos='n'), pos='v'), pos='a')
+                    for token in tokens]
+            cleaned = (" ").join(lemmatized)
+
+        if deromanize:
+            tokens = word_tokenize(cleaned)
+            regex_roman = '^(?=[MDCLXVI])M*(C[MD]|D?C{0,3})(X[CL]|L?X{0,3})(I[XV]|V?I{0,3})$'
+            cleaned = [word for word in tokens if re.search(
+                rex_roman, word, flags=re.IGNORECASE) is None]
+
+        if inplace:
+            self._text = cleaned
+        else:
+            return BookText(filepath=None, rawtext=cleaned)
```

Comment:
This should also pass the meta data to the new `BookText` object.
```suggestion
            return BookText(filepath=None, rawtext=cleaned, author=self.author,
                            title=self.title, meta=self.meta)
```

Reply:
I have responded to this comment in the below commit